### DISCUSSION TITLE
Solved: Failed building on WSL2 (Segmentation fault)

### DISCUSSION THREAD

**Discussion by fzzylogic** [NVlabs/instant-ngp#234] [Q&A]
CUDA: 11.5
OS: Ubuntu 20.04 WSL2

[ 78%] Building CUDA object CMakeFiles/ngp.dir/src/testbed_nerf.cu.o
Segmentation fault (core dumped)
Segmentation fault (core dumped)
make[2]: *** [CMakeFiles/ngp.dir/build.make:258: CMakeFiles/ngp.dir/src/testbed_image.cu.o] Error 139
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/ngp.dir/build.make:244: CMakeFiles/ngp.dir/src/testbed.cu.o] Error 139

Solution was to setup a python3 virtual env and build from within that, ensuring that the 'python' called will be the correct one.

**Comment by bmaltais** [comment]
I am using cuda 11.4 and it is building fine at my end.

**Comment by Evilom** [comment]
when i open fox with GUI in wsl, Segmentation fault too.

**Reply by nelsonzheng** [reply]
This is what I'm getting
```
$ ./build/testbed --scene data/nerf/fox/
03:35:50 INFO     Loading NeRF dataset from
03:35:50 INFO       data/nerf/fox/transforms.json
03:35:51 SUCCESS  Loaded 50 images of size 1080x1920 after 0s
03:35:51 INFO       cam_aabb=[min=[1.0229,-1.33309,-0.378748], max=[2.46175,1.00721,1.41295]]
03:35:51 INFO     Loading network config from: configs/nerf/base.json
03:35:51 INFO     GridEncoding:  Nmin=16 b=1.51572 F=2 T=2^19 L=16
03:35:51 INFO     Density model: 3--[HashGrid]-->32--[FullyFusedMLP(neurons=64,layers=3)]-->1
03:35:51 INFO     Color model:   3--[Composite]-->16+16--[FullyFusedMLP(neurons=64,layers=4)]-->3
03:35:51 INFO       total_encoding_params=13074912 total_network_params=10240
Segmentation fault

**Reply by chenmozxh** [reply]
> when i open fox with GUI in wsl, Segmentation fault too.

same problem.
any solution, thx?
